lib/drivers/bluetoothLE/bluetoothLEDriver.js

```diff
@@ -0,0 +1,196 @@
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import sundial from 'sundial';
+
+import TZOUtil from '../../TimezoneOffsetUtil';
+import annotate from '../../eventAnnotations';
+import { remote } from 'electron';
+
+const isBrowser = typeof window !== 'undefined';
+// eslint-disable-next-line no-console
+const debug = isBrowser ? require('bows')('BluetoothLEDriver') : console.log;
+
+module.exports = (config) => {
+  const cfg = _.clone(config);
+  cfg.deviceTags = ['bgm'];
+
+  // With no date & time settings changes available,
+  // timezone is applied across-the-board
+  cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
+
+  return {
+   /* eslint no-param-reassign:
+      [ "error", { "props": true, "ignorePropertyModificationsFor": ["data"] } ] */
+
+   detect(deviceInfo, cb) {
+     debug('no detect function needed', deviceInfo);
+     cb(null, deviceInfo);
+   },
+
+   setup(deviceInfo, progress, cb) {
+     debug('in setup!');
+     progress(100);
+     cb(null, { deviceInfo });
+   },
+
+   async connect(progress, data, cb) {
+     try {
+       debug('in connect!');
+       await cfg.deviceComms.ble.connectTimeout();
+     } catch (error) {
+       return cb(error, null);
+     }
+     return cb(null, data);
+   },
+
+   async getConfigInfo(progress, data, cb) {
+     debug('in getConfigInfo', data);
+     progress(0);
+
+     try {
+       _.assign(cfg.deviceInfo , await cfg.deviceComms.ble.getDeviceInfo());
+       cfg.deviceTags = ['bgm'];
+       cfg.deviceInfo.deviceId = `${[cfg.deviceInfo.manufacturers]}-${cfg.deviceInfo.model}-${remote.getGlobal('bluetoothDeviceId')}`;
+       cfg.builder.setDefaults({ deviceId: cfg.deviceInfo.deviceId });
+
+       cfg.deviceComms.ble.once('numberOfRecords', (result) => {
+         debug('Number of records:', result);
+       });
+       await cfg.deviceComms.ble.getNumberOfRecords();
+     } catch (error) {
+       return cb(error, null);
+     }
+
+     return cb(null, data);
+   },
+
+   async fetchData(progress, data, cb) {
+     debug('in fetchData', data);
+
+     cfg.deviceComms.ble.once('data', (result) => {
+       debug('Records:', result);
+       data.records = result;
+       return cb(null, data);
+     });
+
+     try {
+       await cfg.deviceComms.ble.getAllRecords();
+     } catch (error) {
+       return cb (error, null);
+     }
+   },
+
+   processData(progress, data, cb) {
+     progress(0);
+     data.post_records = [];
+
+     _.forEach(data.records, (result) => {
+       const record = result;
+       let annotation = null;
+
+       if (record.value > 600) {
+         record.value = 601;
+         annotation = {
+           code: 'bg/out-of-range',
+           value: 'high',
+         };
+       } else if (record.value < 20) {
+         record.value = 19;
+         annotation = {
+           code: 'bg/out-of-range',
+           value: 'low',
+         };
+       }
+
+       if (record.type !== 10) { //check that it's not control solution
```

Comment:
Similar comment here about the value `10` and if this is specific to a certain meter or able to support all sorts of meters.

Reply:
Is there a benefit of uploading control solution records?

Reply:
This does indeed apply to all meters - according to the spec, the type field in the glucose measurement characteristic will be set to 10 if it's control solution.

We would like to upload control solution records in future, but at the moment it is not supported by the backend. We've also not had a single request from users to see control solution records, which probably explains why this hasn't been prioritized on the backend.